Semak Yevgeny

euseand opened the pull request.

euseand committed:
final task initial commit

empty main.py file added

euseand committed:
Added rss_parser.py, test.py with testing console output

euseand committed:
Setup.py file added

euseand committed:
Revert "Setup.py file added"

This reverts commit 0df382778f325ae6dcfe629f3a4ad9b2f3d02448.

euseand committed:
Rss_reader file added

euseand committed:
Revert "Rss_reader file added"

This reverts commit b3eeaa493379c1479c33ea7367240c45f7ff7257.

euseand committed:
Rss_reader and Setup.py files added

euseand committed:
Added serializing to JSON

Added ability to print out news in JSON

AlexeiBuzuma commented on FinalTask/setup.py:
```diff
@@ -0,0 +1,18 @@
+import setuptools
+
+setuptools.setup(
```
Интересная ситуация. 
Не знаю почему, но в чистом докер контейнере с пайтоном пакет почему-то не устанавливается:
```bash
# pip install .
Processing /PythonHomework/FinalTask
Collecting feedparser
  Using cached https://files.pythonhosted.org/packages/91/d8/7d37fec71ff7c9dbcdd80d2b48bcdd86d6af502156fc93846fb0102cb2c4/feedparser-5.2.1.tar.bz2
Collecting bs4
  Using cached https://files.pythonhosted.org/packages/10/ed/7e8b97591f6f456174139ec089c769f89a94a1a4025fe967691de971f314/bs4-0.0.1.tar.gz
ERROR: Could not find a version that satisfies the requirement rss_parser (from euseand-s-rss-parser==0.1) (from versions: none)
ERROR: No matching distribution found for rss_parser (from euseand-s-rss-parser==0.1)
```

AlexeiBuzuma commented on FinalTask/final_task/rss_parser.py:
```diff
@@ -0,0 +1,80 @@
+import feedparser
```
Есть несколько стилистических ошибок
```bash
# pycodestyle --max-line-length=120 .
./setup.py:18:2: W292 no newline at end of file
./final_task/rss_parser.py:80:1: W391 blank line at end of file
./final_task/rss_reader.py:23:32: W292 no newline at end of file
```

AlexeiBuzuma commented on FinalTask/final_task/rss_parser.py:
```diff
@@ -0,0 +1,80 @@
+        self.feed = ''
+        self.news = []
+
+    def parse_rss(self):
```
Возможно эту функцию стоит разбить на несколько небольших функций. 
Она делает очень много вещей и достаточно трудно читается

AlexeiBuzuma commented on FinalTask/final_task/rss_reader.py:
```diff
@@ -0,0 +1,23 @@
+import argparse
+from rss_parser import RssParser
+
+current_version = 0.1
```
точно версия 0.1 ?)

euseand committed:
JSON output bug fixed

euseand committed:
Minor fixes

Added Readme; fixed code style; fixed distribution, now package can be installed properly

euseand committed:
Version fix

euseand committed:
Version fix

euseand committed:
Updated Readme

euseand committed:
Added logging

euseand committed:
Refactored rss_parser.py

euseand committed:
Feature added: iteration 3 partially-implemented

News can be cached and fetched from cache; yet no displayng by date; code refactored in rss_parser.py

euseand committed:
Feature added: 3-rd iteration implemented

euseand committed:
Added docstrings to rss_parser.by

euseand committed:
Added feature: converting news to HTML

euseand committed:
Added feature: converting news to HTML + readme update

euseand committed:
Updated Readme

euseand committed:
Update README.md

euseand committed:
Added feature: convert news to PDF

euseand committed:
Updated readme + gitignore

euseand committed:
Update README.md

euseand committed:
Update README.md

euseand committed:
Feature: iteration 4 fully implemented

euseand committed:
Bug fixed: empty image URLs

euseand committed:
Fixed codestyle issues

euseand committed:
Delete accidental images

AlexeiBuzuma commented:
Есть ошибка, если не передается флаг `--limit`
```bash
# rss-reader https://news.tut.by/rss/world.rss --version
Traceback (most recent call last):
  File "/usr/local/bin/rss-reader", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/rss_reader/rss_reader.py", line 35, in main
    if args.limit > 0:
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

AlexeiBuzuma commented:
при попытке конвертации PDF выдет ошибку 
```bash
rss-reader: rss_parser.py : cache_feed_to_pdf_file : error : TTF Font file not found: /usr/local/lib/python3.8/site-packages/rss_reader/fonts/ttf/DejaVuSerifCondensed.ttf
```

euseand committed:
Bugs fixed: limit flag, pdf convertation

euseand committed:
Added manifest + requirements

euseand committed:
Fixed version flag

euseand committed:
Fixed bugs: caching news from invalid URL, unicode encoding with HTML

euseand committed:
Fixed bug: downloaded images with no JPEG marker

euseand committed:
Bug fixed: prinitng empty news JSON object

AlexeiBuzuma commented on FinalTask/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,127 @@
+current_version = '0.420'
+
+
+def main():
```
Очень большая функция. Ее стоит разбить на несколько маленьких.

AlexeiBuzuma commented on FinalTask/rss_reader/rss_parser.py:
```diff
@@ -0,0 +1,494 @@
+from fpdf import FPDF
+
+
+class RssParser:
```
Данный класс содержит в себе логику не только для парсинга RSS, но так же и для кэширования и конвертирования. Функциональность этого класса стоит разделить на несколько классов.

AlexeiBuzuma commented on FinalTask/rss_reader/rss_parser.py:
```diff
@@ -0,0 +1,494 @@
+                        self.logger.info(f'{total_cached_news} news are in the local cache now')
+        else:
+            print('rss-reader: info : Parse some online news first so there will be something to read from cache')
+            exit()
```
В данном случае бы лучше подошло использование исключений, чем так жестко завершать исполнение программы (в целом лучше функцию `exit` использовать только в тех случаях, когда ты уверен, что она нужна)

AlexeiBuzuma commented on FinalTask/rss_reader/rss_parser.py:
```diff
@@ -0,0 +1,494 @@
+    """
+    Class to parse RSS-news
+    """
+    def __init__(self, url: str, limit: int, verbose: bool, date: str, html_path: str, pdf_path: str):
```
В решении используются тайп хинты. Это плюс :)

AlexeiBuzuma commented on FinalTask/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,127 @@
+import argparse
```
Очень стоило бы реализовать юнит-тесты :)
